crucible-registry.json

```diff
@@ -0,0 +1,14 @@
+{
+    "num-samples": "1",
+    "test-order": "s",
+    "max-sample-failures": "1",
```

Comment:
Hmm, maybe I'm a bit confused...but I thought this registry was supposed to be for the "static" variables as opposed to the "runtime" stuff?

Reply:
These are simply default values that get used if the user does not provide them

Reply:
But what happens when another subproject reads these values instead of the "active" runtime values?

Reply:
I think it depends on the situation.  For the above, these values are only important to rickshaw-run, and so there's not concern about another project reading the original values.  For other values, like roadblock.timeout.default, that is used by multiple subprojects (rickshaw, engine), and those should not be changed.  I guess we need to decide if this registry should be for only things that are shared, or if it should include anything/everything.

Another option may be that something like rickshaw-run copies the registry, alters it, then hands it off (altered) to subprojects it uses.  However, is this that much different than rickshaw-run/result.json?

Reply:
I think maybe we should just have different sections, for example default values that can change at runtime may be in a "defaults" section versus static values that are universal could go in a "static" section or something like that?

I don't really like the idea of creating a copy of the registry for the reasons you allude to.

Reply:
On one hand I am starting to think it might be best to keep it simple, and only store values that do not change and are shared across subprojects.

On the other hand, are we really solving a lot here by not allowing them to be changed (other than a user changing the registry JSON themselves).  What if we really do need user input (for example for a 'crucible run') and that input changes a value that multiple subprojects need (like --max-rb-attempts).  We are still managing the propagation of a lot of these values via cmdline opts for many scripts we call later. 

Perhaps thats just the way it is.  The things in registry are considered sort of global constants, and everything else needs to be managed/passed along via cmdline.

Reply:
I'm not opposed to allowing them to be changed because it would allow us to pass around a lot fewer parameters, but if we do that then we need to come up with some other infrastructure to reset the registry after each run...and decide what, if anything, is persistent, etc.

Reply:
Since pretty much all of this is started from rickshaw, the more I think about this, the more I feel like rickshaw-run.json should just be used to supply all these values to scripts/projects that get called by it.  I wonder if it would clean up an immense amount of cmdline args for launching endpoints, engines, workshop, etc.  The rickshaw-run.json can just go in $run_dir/config/.  Some of the scripts would still need a few cmdline opts, but hopefully not many.

At the end of the day, as long as we have a single source for these values, that's all that should matter.  Getting these hard-coded values out of these scripts is all that we need to do.

Reply:
on the other hand, we have things like this:
`rb_exit_success=0`
`rb_exit_timeout=3`
`rb_exit_abort=4`
`rb_exit_input=2`

Those IMO probably should not be in a rickshaw-run.pl
Maybe the registry is better for things like that.  Things that you would find in a include file

Reply:
Yeah, I feel like we have two different types of info here that we are trying to organize: dynamic runtime parameters and static values that will rarely/never change.  I'm intrigued by the idea of putting the dynamic stuff in the rickshaw-run.json and the static stuff in the registry.